demisto_sdk/commands/split/jsonsplitter.py

```diff
@@ -0,0 +1,125 @@
+        self.new_module_file = new_module_file
+
+        with open(self.input, 'rb') as json_file:
+            self.module_json_data = json.load(json_file)
+
+    def split_json(self):
+        if self.logging:
+            click.secho(f"Starting dashboard extraction from generic module {self.module_json_data.get('name')}",
+                        fg='bright_cyan')
+        self.create_output_dirs()
+        self.create_dashboards()
+        self.create_module()
+        return 0
+
+    def create_output_dirs(self):
+        # no output given
+        if not self.dashboard_dir:
+            # check if in content repository
+            if not is_external_repository() and self.autocreate_dir:
+                pack_name = get_pack_name(self.input)
+                if self.logging:
+                    click.echo(f"No output path given creating dirs in pack {pack_name}")
+                pack_path = os.path.join(PACKS_DIR, pack_name)
+                self.dashboard_dir = os.path.join(pack_path, DASHBOARDS_DIR)
+                self.module_dir = os.path.join(pack_path, GENERIC_MODULES_DIR)
+
+                # create the dirs, dont fail if exist
+                try:
+                    os.mkdir(self.dashboard_dir)
+                except FileExistsError:
+                    pass
+
+                try:
+                    os.mkdir(self.module_dir)
+                except FileExistsError:
+                    pass
+
+            # if not in content create the files locally
+            else:
+                if self.logging:
+                    click.echo("No output path given and not running in content repo, creating files locally")
+                self.dashboard_dir = '.'
+                self.module_dir = '.'
+
+    def create_dashboards(self):
+        if self.logging:
+            click.echo("Starting dashboard creation")
+
+        for view in self.module_json_data.get('views', []):
+            for tab in view.get('tabs', []):
+                dashboard_data = tab.get('dashboard')
+
+                if dashboard_data:
+                    dashboard_file_name = dashboard_data.get('name').replace(' ', '') + '.json'
+                    full_dashboard_path = os.path.join(self.dashboard_dir, dashboard_file_name)
+
+                    if self.logging:
+                        click.echo(f"Creating dashboard: {full_dashboard_path}")
+
+                    with open(full_dashboard_path, 'w') as dashboard_file:
+                        json.dump(dashboard_data, dashboard_file, indent=4)
+
+                    tab['dashboard'] = {'id': dashboard_data.get('id')}
+
+    def create_module(self):
+        if not self.new_module_file:
+            if self.logging:
+                click.echo(f"Updating module file {self.input}")
+            module_file_path = self.input
+
+        else:
+            if self.logging:
+                click.echo("Creating new module file")
+            file_name = os.path.split(self.input)[1]
```

Comment:
I think it will be more clear this way
```suggestion
            file_name = os.path.basename(self.input)
```